src/test/java/org/cryptomator/cryptofs/fh/ChunkCacheTest.java

```diff
@@ -20,147 +25,170 @@ public class ChunkCacheTest {
+	@DisplayName("getChunk invokes chunkLoader.load() on cache miss")
 	public void testGetInvokesLoaderIfEntryNotInCache() throws IOException, AuthenticationFailedException {
 		long index = 42L;
-		Chunk chunk = mock(Chunk.class);
-		when(chunkLoader.load(index)).thenReturn(chunk);
+		var data = ByteBuffer.allocate(0);
+		when(chunkLoader.load(index)).thenReturn(data);
+
+		inTest.getChunk(index);
 
-		Assertions.assertSame(chunk, inTest.get(index));
+		verify(chunkLoader).load(index);
 		verify(stats).addChunkCacheAccess();
 	}
 
 	@Test
+	@DisplayName("getChunk does not invoke chunkLoader.load() on cache hit due to getting chunk twice")
 	public void testGetDoesNotInvokeLoaderIfEntryInCacheFromPreviousGet() throws IOException, AuthenticationFailedException {
 		long index = 42L;
-		Chunk chunk = mock(Chunk.class);
-		when(chunkLoader.load(index)).thenReturn(chunk);
-		inTest.get(index);
+		var data = ByteBuffer.allocate(0);
+		when(chunkLoader.load(index)).thenReturn(data);
+
+		var chunk = inTest.getChunk(index);
+		var sameChunk = inTest.getChunk(index);
 
-		Assertions.assertSame(chunk, inTest.get(index));
+		Assertions.assertSame(chunk, sameChunk);
+		verify(chunkLoader, Mockito.times(1)).load(index);
 		verify(stats, Mockito.times(2)).addChunkCacheAccess();
-		verify(chunkLoader).load(index);
+		verify(stats, Mockito.times(1)).addChunkCacheMiss();
 	}
 
 	@Test
+	@DisplayName("getChunk does not invoke chunkLoader.load() on cache hit due to getting after putting")
 	public void testGetDoesNotInvokeLoaderIfEntryInCacheFromPreviousSet() throws IOException {
 		long index = 42L;
-		Chunk chunk = mock(Chunk.class);
-		inTest.set(index, chunk);
+		var data = ByteBuffer.allocate(0);
+		var chunk = inTest.putChunk(index, data);
 
-		Assertions.assertSame(chunk, inTest.get(index));
+		Assertions.assertSame(chunk, inTest.getChunk(index));
 		verify(stats).addChunkCacheAccess();
 	}
 
 	@Test
-	public void testGetInvokesSaverIfMaxEntriesInCacheAreReachedAndAnEntryNotInCacheIsRequested() throws IOException, AuthenticationFailedException {
-		long firstIndex = 42L;
-		long indexNotInCache = 40L;
-		Chunk chunk = mock(Chunk.class);
-		inTest.set(firstIndex, chunk);
-		for (int i = 1; i < ChunkCache.MAX_CACHED_CLEARTEXT_CHUNKS; i++) {
-			inTest.set(firstIndex + i, mock(Chunk.class));
-		}
-		when(chunkLoader.load(indexNotInCache)).thenReturn(mock(Chunk.class));
-
-		inTest.get(indexNotInCache);
+	@DisplayName("putChunk returns a dirty chunk")
+	public void testPutChunkReturnsDirtyChunk() {
+		long index = 42L;
+		var data = ByteBuffer.allocate(0);
+		var chunk = inTest.putChunk(index, data);
 
-		verify(stats).addChunkCacheAccess();
-		verify(chunkSaver).save(firstIndex, chunk);
-		verify(bufferPool).recycle(chunk.data());
-		verifyNoMoreInteractions(chunkSaver);
+		Assertions.assertTrue(chunk.isDirty());
 	}
 
-	@Test
-	public void testGetInvokesSaverIfMaxEntriesInCacheAreReachedAndAnEntryNotInCacheIsSet() throws IOException {
+	@RepeatedTest(100)
```

Comment:
Why do we need a repeated test here?

Reply:
With the old implementation I noticed that it failed on CI but succeeded on my local machine. I then noticed that the implementation was timing-sensitive: If you remove the `cdl.await()`, you'll see that `chunkSaver.save(...)` is not called immediately but in an async maintenance thread (as mentioned in the Caffeine docs). Therefore the verifications in this tests failed _sometimes_.